tests/Jackalope/Transport/DoctrineDBAL/XmlPropsRemover/XmlPropsRemoverTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Jackalope\Transport\DoctrineDBAL\XmlPropsRemover;
+
+use Jackalope\Factory;
+use Jackalope\Test\TestCase;
+use Jackalope\Transport\DoctrineDBAL\XmlParser\XmlToPropsParser;
+use PHPCR\Util\ValueConverter;
+
+class XmlPropsRemoverTest extends TestCase
+{
+    public function testRemoveProps(): void
+    {
+        $xml = <<<EOT
+<?xml version="1.0" encoding="UTF-8"?>
+<sv:node xmlns:mix="http://www.jcp.org/jcr/mix/1.0" xmlns:nt="http://www.jcp.org/jcr/nt/1.0" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:jcr="http://www.jcp.org/jcr/1.0" xmlns:sv="http://www.jcp.org/jcr/sv/1.0" xmlns:rep="internal">
+	<sv:property sv:name="jcr:primaryType" sv:type="Name" sv:multi-valued="0">
+		<sv:value length="15">nt:unstructured</sv:value>
+	</sv:property>
+	<sv:property sv:name="jcr:mixinTypes" sv:type="Name" sv:multi-valued="1">
+		<sv:value length="9">sulu:page</sv:value>
+	</sv:property>
+	<sv:property sv:name="jcr:uuid" sv:type="String" sv:multi-valued="0">
+		<sv:value length="36">0804f0c3-5250-4c2f-9d7e-7d0c99103026</sv:value>
+	</sv:property>
+	<sv:property sv:name="i18n:en-title" sv:type="String" sv:multi-valued="0">
+		<sv:value length="8">My Title</sv:value>
+	</sv:property>
+	<sv:property sv:name="ampersand" sv:type="String" sv:multi-valued="0"><sv:value length="13">foo &amp; bar&amp;baz</sv:value></sv:property>
+	<sv:property sv:name="äüö?ß&lt;&gt;''&quot;=&quot;test" sv:type="String" sv:multi-valued="0"><sv:value length="15">&lt;&gt;:&amp;|öäü"?"ß'='</sv:value></sv:property>
+	<sv:property sv:name="block_1_ref" sv:type="reference" sv:multi-valued="0">1922ec03-b5ed-40cf-856c-ecfb8eac12e2</sv:property>
```

Comment:
ah, actually @alexander-schranz are you sure that this is correct? should it not be `Reference` starting with capital `R`?

we might need to backport the strtolower to 1.x too

Reply:
I think it would make sense to backport the strtolower.

Reply:
this seems to have been a mess already before: https://github.com/jackalope/jackalope-doctrine-dbal/blob/1.x/src/Jackalope/Transport/DoctrineDBAL/Client.php#L746

i tried to find the system view xsd definition but the URI does not lead to a download link. and the whole jackrabbit source code neither contains the schema nor links to it with schemaLocation :-(

i will backport the fix and see if there are other places with invalid assumptions.

Reply:
interesting then it does also make more sense to use strtolower here.

Reply:
i now use the PHPCR PropertyType static methods to avoid case errors.